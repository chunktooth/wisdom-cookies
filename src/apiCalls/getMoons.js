/*eslint max-len: ["error", { "ignoreUrls": true, 
"ignoreTemplateLiterals": true }]*/

import { clientID, clientSecret } from '../apiKey.js';

export const getMoons = async () => {
  let root = 'https://api.aerisapi.com/';

  try {
    const response = await fetch(`${root}sunmoon/moonphases?limit=4&client_id=${clientID}&client_secret=${clientSecret}`);
    const moonphases = await response.json();
    
    return moonphases;
  } catch (error) {
    throw new Error(`Caught error: ${error.message}`);
  }
};


