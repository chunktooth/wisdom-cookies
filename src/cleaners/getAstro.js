/*eslint max-len: ["error", { "ignoreUrls": true }]*/
// export const getAstro = async () => {
//   try {
//     const response = await fetch('https://aztro.sameerkumar.website/?sign=&day=today');
//     const today = await response.json();
//     console.log(today);

//     return today;
//   } catch (error) {
//     throw new Error(`Error found: ${error.message}`);
//   }
// };