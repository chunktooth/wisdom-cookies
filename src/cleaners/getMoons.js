/*eslint max-len: ["error", { "ignoreUrls": true }]*/
export const getMoons = async () => {
  let unixTime = Date.now();

  try {
    const response = await fetch(`http://farmsense-prod.apigee.net/v1/moonphases/?d=${unixTime}`);
    const moonphases = await response.json();
  
    return moonphases;
  } catch (error) {
    throw new Error(`Caught error: ${error.message}`);
  }
};