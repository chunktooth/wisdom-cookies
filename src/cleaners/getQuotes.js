/*eslint max-len: ["error", { "ignoreUrls": true }]*/
export const getQuotes = async () => {
  try {
    const response = await fetch('http://fortunecookieapi.herokuapp.com/v1/fortunes?limit=1000&skip=&page=');
    const quotes = await response.json();
    console.log(quotes)
    
    const wisdom = await getWisdom(quotes);
   
    return wisdom;
  } catch (error) {
    throw error;
  }
};

const getWisdom = async (quotes) => {
  const randomKey = Math.floor(Math.random() * quotes.length);
  const wisdom = quotes[randomKey];

  return wisdom;
}