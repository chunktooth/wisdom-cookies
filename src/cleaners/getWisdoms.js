/*eslint max-len: ["error", { "ignoreUrls": true }]*/
export const getWisdoms = async () => {
  try {
    const response = await fetch('http://fortunecookieapi.herokuapp.com/v1/fortunes?limit=1000&skip=&page=');
    const wisdoms = await response.json();
   
    return wisdoms;
  } catch (error) {
    throw error;
  }
};