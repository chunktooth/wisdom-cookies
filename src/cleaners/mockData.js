/* exported variableName */
export const mockQuotes = [
  {
    'message': 'Mankind is unkind, man',
    'id': '5280'
  },
  {
    'message': 'Godluck and Goodspeed!',
    'id': '1234'
  }
];