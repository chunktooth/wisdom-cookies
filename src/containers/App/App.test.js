import React from 'react';
import { App, 
  mapStateToProps, 
  mapDispatchToProps } from './App';
import { shallow } from 'enzyme';
import { mockJar, 
  mockWisdoms,
  mockMoonphases } from '../../mockData';
import { getWisdoms } from '../../cleaners/getWisdoms';
import { getMoons } from '../../cleaners/getMoons';
jest.mock('../../cleaners/getWisdoms');
jest.mock('../../cleaners/getMoons');

describe('App', () => {
  let wrapper;
  const mockLoadWisdoms = jest.fn();
  const mockLoadMoons = jest.fn();
    
  beforeEach(() => {
    wrapper = shallow(<App 
      jar={mockJar}
      loadWisdoms={mockLoadWisdoms} 
      loadMoons={mockLoadMoons} />,
    { disableLifecycleMethods: true });
  });

  it.skip('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should fetch and load wisdoms into store', async () => {
    await wrapper.instance().componentDidMount();
    await expect(getWisdoms).toHaveBeenCalled();
  });

   it('should fetch and load moons into store', async () => {
    await wrapper.instance().componentDidMount();
    await expect(getMoons).toHaveBeenCalled();
  });

  it('should map jar state to props', () => {
    const mappedJar = mapStateToProps(mockJar);
    expect(mappedJar.jar).toEqual(mockJar.jar);
  });

  it('should map wisdoms state to props', () => {
    const mappedWisdoms = mapStateToProps(mockWisdoms);
    expect(mappedWisdoms.wisdoms).toEqual(mockWisdoms.wisdoms);
  });

  it('should map moonphases state to props', () => {
    const mappedMoonphases = mapStateToProps(mockMoonphases);
    expect(mappedMoonphases.moonphases).toEqual(mockMoonphases.moonphases);
  });

  it('should dispatch loadWisdoms to props', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.loadWisdoms();
    expect(mockDispatch).toHaveBeenCalled();
  });

  it('should dispatch loadMoons to props', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.loadMoons();
    expect(mockDispatch).toHaveBeenCalled();
  });

});