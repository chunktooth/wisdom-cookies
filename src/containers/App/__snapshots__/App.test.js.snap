// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    jar={
        Array [
            Object {
              "id": "5280",
              "message": "Mankind is unkind, man",
            },
            Object {
              "id": "1234",
              "message": "Godluck and Goodspeed!",
            },
          ]
    }
    loadMoons={[Function]}
    loadWisdoms={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="left-column"
>
          <header
                    className="App-header"
          >
                    <h1
                              className="title"
                    >
                              Wi$dom Co0kies
                    </h1>
                    <h4
                              className="under-title"
                    >
                              Choose wisdom to guide your fortune
                    </h4>
          </header>
          <img
                    alt="A Chinese dragon soaring"
                    className="dragon"
                    src="dragon.png"
          />
          <Connect(Cookie) />
</div>,
        <div
          className="right-column"
>
          <Connect(Jar) />
</div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header
              className="App-header"
>
              <h1
                            className="title"
              >
                            Wi$dom Co0kies
              </h1>
              <h4
                            className="under-title"
              >
                            Choose wisdom to guide your fortune
              </h4>
</header>,
            <img
              alt="A Chinese dragon soaring"
              className="dragon"
              src="dragon.png"
/>,
            <Connect(Cookie) />,
          ],
          "className": "left-column",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="title"
>
                  Wi$dom Co0kies
</h1>,
                <h4
                  className="under-title"
>
                  Choose wisdom to guide your fortune
</h4>,
              ],
              "className": "App-header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Wi$dom Co0kies",
                  "className": "title",
                },
                "ref": null,
                "rendered": "Wi$dom Co0kies",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Choose wisdom to guide your fortune",
                  "className": "under-title",
                },
                "ref": null,
                "rendered": "Choose wisdom to guide your fortune",
                "type": "h4",
              },
            ],
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "A Chinese dragon soaring",
              "className": "dragon",
              "src": "dragon.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Jar) />,
          "className": "right-column",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="left-column"
>
            <header
                        className="App-header"
            >
                        <h1
                                    className="title"
                        >
                                    Wi$dom Co0kies
                        </h1>
                        <h4
                                    className="under-title"
                        >
                                    Choose wisdom to guide your fortune
                        </h4>
            </header>
            <img
                        alt="A Chinese dragon soaring"
                        className="dragon"
                        src="dragon.png"
            />
            <Connect(Cookie) />
</div>,
          <div
            className="right-column"
>
            <Connect(Jar) />
</div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <header
                className="App-header"
>
                <h1
                                className="title"
                >
                                Wi$dom Co0kies
                </h1>
                <h4
                                className="under-title"
                >
                                Choose wisdom to guide your fortune
                </h4>
</header>,
              <img
                alt="A Chinese dragon soaring"
                className="dragon"
                src="dragon.png"
/>,
              <Connect(Cookie) />,
            ],
            "className": "left-column",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="title"
>
                    Wi$dom Co0kies
</h1>,
                  <h4
                    className="under-title"
>
                    Choose wisdom to guide your fortune
</h4>,
                ],
                "className": "App-header",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Wi$dom Co0kies",
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": "Wi$dom Co0kies",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Choose wisdom to guide your fortune",
                    "className": "under-title",
                  },
                  "ref": null,
                  "rendered": "Choose wisdom to guide your fortune",
                  "type": "h4",
                },
              ],
              "type": "header",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "A Chinese dragon soaring",
                "className": "dragon",
                "src": "dragon.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Jar) />,
            "className": "right-column",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
